src/types.ts

```diff
@@ -45,85 +36,88 @@ export type Group = {
         title?: string; // If not provided, will attempt to deduce.
       };
     }
+);
+
+// Object Endpoints Types
+export type GetObjectRequest = {
+  objectId: string;
+};
+
+export type ListObjectsRequest = {
+  parentId?: string;
+  limit?: number;
+  startingAfter?: string;
+  endingBefore?: string;
+};
+
+export type ListObjectsResponse = {
+  objects: Object[];
+  hasMore: boolean;
+};
+
+export type CreateObjectRequest = {
+  parentId?: string;
+  properties?: ObjectProperties;
+} & (
   | {
-      kind: 'email';
-      metadata: {
-        email: string;
-        sent?: Date; // If not provided, will attempt to deduce.
-        from?: string; // If not provided, will attempt to deduce.
-        subject?: string; // If not provided, will attempt to deduce.
-        to?: string[]; // If not provided, will attempt to deduce.
-      };
-    }
-  | {
-      kind: 'notion_page';
-      metadata: {
-        pageId: string;
-        url: string;
-        title?: string;
-      };
-    }
-  | {
-      kind: 'epub';
-      metadata: {
-        epubUrl: string;
-        title?: string;
-        language?: string;
-      };
+      type: 'collection';
+      metadata: {};
     }
   | {
-      kind: 'notion';
+      type: 'text';
       metadata: {
-        accessToken: string;
+        text: string;
       };
     }
   | {
-      kind: 'github_repository';
+      kind: 'html';
       metadata: {
-        accessToken: string;
-        repoOwner: string;
-        repoName: string;
-        rootPath: string;
+        html: string;
+        title?: string; // If not provided, will attempt to deduce.
       };
     }
   | {
-      kind: 'pdf';
+      kind: 'markdown';
       metadata: {
-        pdfUrl: string;
+        markdown: string;
+        title?: string; // If not provided, will attempt to deduce.
       };
     }
 );
 
+export type DeleteObjectRequest = {
+  objectId: string;
+};
+
+export type DeleteObjectResponse = {
+  deleted: boolean;
+};
+
+// Operation Endpoints Types
+
 export type Atom = {
   id: string;
   createdAt: Date;
   updatedAt: Date;
-  groupId: string;
+  objectId: string;
   content: string;
-  kind: 'title' | 'content' | 'link' | 'image' | 'code';
+  type: 'title' | 'content' | 'link' | 'image' | 'code' | 'list_item';
 };
```

Comment:
Hmm, we shouldn't expose atoms here since they aren't super useful (and frankly, more-so internal). Let's change this into a "snippet" (sidenote: open to naming suggestions) with two fields: `content` and `type`. This will likely require backend changes too.

Once this is done, we no longer have to return atoms from content search results.